r/R/arrow-package.R

```diff
@@ -76,6 +76,39 @@ option_use_threads <- function() {
   !is_false(getOption("arrow.use_threads"))
 }
 
+#' Report information on the package's capabilities
+#'
+#' This function summarizes a number of build-time configurations and run-time
+#' settings for the Arrow package. It may be useful for diagnostics.
+#' @return A list including version information, boolean "capabilities", and
+#' statistics from Arrow's memory allocator.
```

Comment:
Does it make sense to include runtime statistics together with static configuration information?

Reply:
I don't think it makes sense to separate them; I expect that this is a function I'll ask bug reporters to run to give me all of the arrow debug information about their setup.